tests/subsys/event_manager/src/main.c

```diff
@@ -64,6 +73,107 @@ static void test_multicontext(void)
 	test_start(TEST_MULTICONTEXT);
 }
 
+static void test_event_size_static(void)
+{
+	if (!IS_ENABLED(CONFIG_EVENT_MANAGER_PROVIDE_EVENT_SIZE)) {
+		ztest_test_skip();
+		return;
+	}
+
+	struct test_size1_event *ev_s1;
+	struct test_size2_event *ev_s2;
+	struct test_size3_event *ev_s3;
+	struct test_size_big_event *ev_sb;
+
+	ev_s1 = new_test_size1_event();
+	zassert_equal(sizeof(*ev_s1), event_manager_event_size(&ev_s1->header),
+		"Event size1 unexpected size");
+	event_manager_free(ev_s1);
+
+	ev_s2 = new_test_size2_event();
```

Comment:
I think it would be good to update the OOM error handler when we allocate events. Current implementation may lead to some unexpected problems (this fix does not have to be addressed as part of this PR).

Reply:
This should be fixed under a separate ticket (NCSDK-13943)

Reply:
So leaving it for future fix.